src/Components/SystemDetail/SystemDetail.js

```diff
@@ -12,18 +13,39 @@ const SystemRecommendations = React.lazy(() => import('./SystemRecommendations')
 
 const SystemDetail = (props) => (
     <Suspense fallback="">
-        <SystemRecommendations inventoryId={ props.rosSystemInfo.inventory_id }/>
+        <SystemRecommendations inventoryId={ props.entity.id }/>
     </Suspense>
 );
 
 const mapStateToProps = (state) => {
     return {
-        rosSystemInfo: state.systemDetailReducer?.systemInfo
+        entity: state.entityDetails && state.entityDetails.entity
     };
 };
 
 SystemDetail.propTypes = {
-    rosSystemInfo: PropTypes.object
+    entity: PropTypes.object
+};
+
+const ConnectedSystemDetail = connect(mapStateToProps, null)(SystemDetail);
+
+const SystemDetailWrapper = ({ getRegistry, ...props }) => {
+    const [Wrapper, setWrapper] = useState();
+    useEffect(() => {
+        if (getRegistry) {
+            getRegistry()?.register?.({ systemRecsReducer });
+        }
+
+        setWrapper(() => getRegistry ? Provider : Fragment);
+    }, [getRegistry]);
```

Comment:
Here, I was getting a lint warning `useEffect missing dependency`  so I passed `getRegistry` as dependency to `useEffect()`